test/framework/openshift_resources.go

```diff
@@ -91,6 +99,23 @@ func checkResourcesInPods(namespace string, dcName string, expected corev1.Resou
 	}
 }
 
+func checkResourcesInPodsByDeployment(namespace string, dName string, expected corev1.ResourceRequirements) (bool, error) {
+	pods, err := GetPodsByDeployment(namespace, dName)
```

Comment:
What about to keep the checkResourcesInPod method and instead of having the namespace and dName (or dcName), we provide a function to get the pods. This way we don't need to duplicate this method. Something like:

```go
checkResourcesInPods(getPodsByDeployment(namespace, dName), expected) // for deployment
checkResourcesInPods(getPodsByDeploymentConfig(namespace, dcName), expected) // for deploymentConfig
```

Reply:
Will take different approach in the end, will get list of pods directly in `WaitForPodsByxxxToHaveResources` and pass here directly list of pods.